### ISSUE TITLE
Release SecureDrop 0.3.7

### ISSUE THREAD

**Issue by garrettr** [freedomofpress/securedrop#1312]
This is a tracking issue for the final tasks involved in releasing SecureDrop 0.3.7. We'll use a checklist in this top comment to track the remaining work - note that it may change over time, as new items are added, removed, or split into subtasks.

# Finish release candidate (0.3.7+rc1)

- [x] Merge open pull requests from 0.3.7 milestone - @garrettr
- [x] Build test debs - @garrettr
- [x] Stand up test apt server with 0.3.7+rc1 packages - @ageis 

# QA

## General testing for a new release

- [x] Test new install of 0.3.7 - @garrettr
- [x] Test automatic upgrade from 0.3.6 to 0.3.7 - @garrettr
- [x] Test automatic upgrade from 0.3.5 to 0.3.7 (since there may still be some instances that are stuck on 0.3.5 due to the expired code signing key). - @garrettr

## Release-specific tests

- [x] ~~Test backup and restore on a full production+hardware installation of SecureDrop.~~
 - We discussed this and decided that this has been sufficiently tested on virtualized production environments and does not need additional testing on a hardware installation of SecureDrop.
- [x] Test Tails 1.x -> 2.x upgrade procedure and documentation. - @conorsch 
 - [x] Document how to upgrade `tails_files` on Tails workstations that have been upgraded from Tails 1.x to 2.x. This has many of the same steps as the [upgrade procedure](https://docs.securedrop.org/en/latest/upgrade/0.3.5-to-0.3.6.html#upgrade-procedure) for 0.3.5 to 0.3.6, so you can crib from there. - @garrettr
 - [x] File follow-up issue for documenting "how to upgrade SecureDrop on your Tails workstation" steps in a designated location (since it is a common workflow for admins). - @garrettr

# Release

## 0.3.6

We still need to release the "no-op" Debian packages for 0.3.6. 0.3.6 only included changes to the Ansible playbooks (to update the expiration date on the code signing key), so the Debian packages for 0.3.6 are functionally identical to those for 0.3.5 (except they bump the version number).

We haven't released these packages yet because we plan to use them in conjunction with monitoring to determine which instances have not yet done the manual upgrade for 0.3.6.

Related issue: #1218

- [x] Build final Debian packages for 0.3.6
- [x] Release 0.3.6
- [x] Publish blog post about 0.3.6 Debian package release

## 0.3.7

- [x] Build final Debian packages for 0.3.7
- [x] Release 0.3.7
- [x] Publish blog post about 0.3.7 Debian package release

# Post-release

- [ ] Merge release changes into development branch

**Comment by conorsch** [comment]
Great overview of the next few days. A note about the backup and restore testing: let's chain the 0.3.6 upgrade and 0.3.7 restore tasks on hardware, so:

1. allow existing 0.3.6 hardware instance to autoupgrade to 0.3.6 deb packages (so Source interface displays 0.3.6)
2. checkout the 0.3.7 rc branch and run the new backup role, storing on a separate usb stick
3. perform a fresh install of 0.3.7 on hardware
4. restore submissions from prior backup, using new 0.3.7 restore logic

The fresh install will also be a fine test of the new tails files changes, since most of that testing has been around upgrades recently.

**Comment by garrettr** [comment]
For this release, we are going "squash-and-merge" our pull requests. This makes the history on the release branch easier to understand, and will make it _much_ easier for us to cherry-pick the changes onto the development branch.

The order in which the pull requests are merged is slightly important. After some review, I think the best order is:

- [ ] #1273
- [ ] #1309 - Fixes Tails scripts so they work on 2.x
 - [ ] #1303 - Documents how to upgrade from 1.x to 2.x (made possible by #1309)
- [ ] #1310 (needed to use automated tests for #1306)
 - [ ] #1306

**Comment by ageis** [comment]
I'm ready with a test server for the debs when you are. I presume for the auto-update testing that we will stand up a 0.3.6 instance that uses an alternate `apt_repo_url` variable. We may as well do that before we push the debs. Consider the URL to be `apt-test.freedom.press`.

**Comment by garrettr** [comment]
@ageis That's correct. We'll also want to generate a GPG signing key for testing. It's ok to generate it on the test server and let aptly sign the packages automatically (unlike we do in production, where we sign with an offline key).

**Comment by garrettr** [comment]
Testing debs: [securedrop-0.3.7+rc1-debs.tar.gz](https://github.com/freedomofpress/securedrop/files/287353/securedrop-0.3.7.rc1-debs.tar.gz)

**Comment by ageis** [comment]
SecureDrop 0.3.7+rc1 debs are live on apt-test.freedom.press. Here's the test signing key that was used to sign the testing repository (expires in one month).
```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: GnuPG v1

mQENBFdIucQBCADMR103mZKYK5eHgeBXpf7VTOyLWKDTHloU22USb4m3NWsl42B4
uLn3LLMQxdsIMNrJ0T6hdT8rG88pj972cgdjdUj8Ol+0G36bWz0POCToj6LyhMQU
TY+n477nlEJgkrLPmkNmiUeSU8bpvWG0ZrZYp1A3R9iwAr6OgEcFVXMegNhC/8AO
JBOt6gRj108Z8L6nBWz611rSlH4hng8uTT9h6weQd1ovawB5tFAkjZTGK/uGBbxD
+Q6yYimB1QebIAbKyMKaC107ulK0rJeBxy2TeQqWRYjJAgeUkigHXq80B9gYTce7
cAYRSyfL0kVPSjghfLKYncGeK5JXUWC209yJABEBAAG0OkZyZWVkb20gb2YgdGhl
IFByZXNzIEZvdW5kYXRpb24gVGVzdCBTaWduaW5nIEtleSAoVEVTVElORymJAT4E
EwECACgFAldIucQCGwMFCQAnjQAGCwkIBwMCBhUIAgkKCwQWAgMBAh4BAheAAAoJ
EOpnEVFyl6rPCdYH+QGiw3JZ6OCzbTFaEHicMugfwUk7j2LnHgKTloK1wZlleIos
CsHgP7osIu/lakAjsZs4p7l04YQLYi9L7Q18C8/D5aCjpZYfQXybeZ7uF/FOTEcg
I6V0b68QGos3WzHhmLWaVVKpIKmk1FgkpuSXGJtVwsHqLfj7KOFpfdIS+3H+tJ2D
qXh7sz9w+XIXq99SRTvzQauCNwetwmrx9rD8ZulFuImrEv4Drw3HGfoN8u13lAJq
AcnL51rx7G02ZIvQqfpOS+zKw2tqBS6Qd1HyUjOZy6tW5e0ZHFZ1482VVgpXWENy
k3hIM3hlMfTnwCILzCdqbpxp9uTOxTrRucNBHL65AQ0EV0i5xAEIAOlxpxjpoY3M
PPtf22w0+elwpBn0/0UepKoYC+0OBJSODlkuJKE6AaLxnMJoOyJx3MhjNbzx5mU1
Sc9j3mRsNDaGnlsfunz6gl2FOrnmORAN2ToNxfIkKRf1adMVTXiy1XEIVXz08MUT
bgo34fefjYmxUx4hMnQXov1zDCLEayMY5zra23XOHn4B0TGWhNUFkn8vy2eCzAPw
0Frp3aMIHwJINQc0xhi3qvtFnm4L9TR6kbsNT5qO5LXa8eFYe6YHudLyxJ3ejZuQ
Eo1Cz7phVztEf/G3bpfl00/iZUjzQLJiBtKPSkbxkf3gNfG3EtvKVt2nlJ6rEVOD
GPAROC1aRJEAEQEAAYkBJQQYAQIADwUCV0i5xAIbDAUJACeNAAAKCRDqZxFRcpeq
zzMuCACGt7c8P7LTE5PGU+t6z6NNL4E753AezH1R2iNqa6rljBwS70sj6+75l69v
N+UUxftVPP7uaSdxBze9mcTBZ4usgOhdS6HPXxlke4iWrH2/+TpFEo09KxfpOz9W
eC0h/B+AoIB+7ZODw2g3Gsrd1m2EDZny1gYac/CDjURcq+ewqMO2XZE7wTeLhxdw
bW0W7IbUMVQqK2ncOXsqJES0hMWiHdoB+j4tqDPdfvooBzFH1Y9Y+8LDbQYF6UfQ
qabCmO8OwSYpYm0btqimTfKQzNudIpdde1zctlPgV4CYqKnmxrLHXugwag/77RfH
t2C7C0iOep3FPPsi1HP1Zyk1iV3O
=G2em
-----END PGP PUBLIC KEY BLOCK-----
```

**Comment by garrettr** [comment]
We've assigned the final Release QA testing tasks to individuals.

If you encounter an issue in 0.3.7+rc1 during testing, please file an issue on the **0.3.7** milestone with the **QA: Release** label so we can better keep track of them.

**Comment by ageis** [comment]
Our monitoring (via Nagios) tracks 26 instances in the SecureDrop ecosystem. Parallel to the normal service uptime checks, there is a check that monitors for the presence of a version string, which only alerts FPF internally. When we release the 0.3.6 .deb packages (and later, 0.3.7) on our repository, we'll be able to see which instances upgraded overnight and which haven't. 

As of now the upgrade monitoring has been bumped to look for version 0.3.6 instead of 0.3.5. I'm standing by ready to publish the .debs and hand @garrettr the Release file for signing.

**Comment by garrettr** [comment]
The 0.3.6 debs have been released and we are tracking successful upgrades with Nagios to get a sense of how many instances successfully did the manual upgrade process for 0.3.6 (this process had to be manual in order to update the expired package signing key). Once that is done, and the last of the QA testing for the release is finished, we will be ready to release 0.3.7 - hopefully by tomorrow afternoon!

**Comment by garrettr** [comment]
The 0.3.6 release announcement blog post has been published: https://securedrop.org/news/securedrop-036-released.

**Comment by conorsch** [comment]
SecureDrop 0.3.7 was released on 2016-06-02. Backporting of release changes onto develop is tracked in #1313, so closing this issue.

**Comment by garrettr** [comment]
@conorsch FWIW, this issue was intentionally left open because there is still an extant unfinished task in the Post-release section of the original release checklist: "Merge release changes into development branch". You're correct that this work is tracked in #1313, but my intention was to use this issue as a master tracking issue for the entire release; it would not be closed until *all* work was completed. The 0.3.7 fulfills the same function though, so I'm ok with it being closed.
